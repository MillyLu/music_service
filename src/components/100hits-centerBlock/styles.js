import styled from "styled-components";

export const MainCenterblock = styled.div`
width: auto;
flex-grow: 3;
padding: 20px 40px 20px 111px;
background-color: ${props => props.theme === "light" ? "#FFFFFF" : "#1C1C1C"}; 
`;

export const CenterblockContent = styled.div`
display: flex;
flex-direction: column;
width: 1107px;
`;