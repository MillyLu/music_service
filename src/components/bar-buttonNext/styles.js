import styled from "styled-components";
import {ReactComponent as NextSvg} from '../svg/next/next.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonNext = styled(PlayerButton)`
margin-right: 28px;
fill: #a53939;
`;

export const PlayerButtonNextSvg = styled(NextSvg)`
width: 15px;
height: 14px;
fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
 & :hover {
    fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
 }
`;