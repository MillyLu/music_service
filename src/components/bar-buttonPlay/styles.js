import styled from "styled-components";

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const ButtonCursor = styled(PlayerButton)`
cursor: pointer;
`

export const PlayerButtonPlay = styled(ButtonCursor)`
margin-right: 23px;
`;

export const PlayerButtonPlaySvg = styled.svg`
width: 22px;
  height: 20px;
  fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
  & :hover {
     fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
  }
`;

export const Audio = styled.audio`
display: none;
`