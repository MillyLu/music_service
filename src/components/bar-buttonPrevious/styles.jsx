import styled from "styled-components";
import {ReactComponent as PrevSvg} from '../svg/prev/prev.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonPrev = styled(PlayerButton)`
margin-right: 23px;
`;

export const PlayerButtonPrevSvg = styled(PrevSvg)`
width: 15px;
height: 14px;
fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
 & :hover {
    fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
 }
`;