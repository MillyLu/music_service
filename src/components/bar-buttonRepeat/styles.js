import styled from "styled-components";
import {ReactComponent as RepeatSvg} from '../svg/repeat/repeat.svg';


export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonIcon = styled(PlayerButton)``;

export const PlayerButtonRepeat = styled(PlayerButtonIcon)`
margin-right: 24px;
`;

export const PlayerButtonRepeatSvg = styled(RepeatSvg)`
width: 18px;
height: 12px;
fill: none;
stroke: ${props => props.theme === "light" ? "#B1B1B1" : "#696969"};

${PlayerButtonRepeat}:hover & {
    fill: none;
    stroke: ${props => props.theme === "light" ? "#707070" : "#ACACAC"};
    cursor: pointer;
  }
${PlayerButtonRepeat}:active & {
    fill: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    stroke: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    cursor: pointer;
  }
`;