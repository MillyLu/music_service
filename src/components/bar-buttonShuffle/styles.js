import styled from "styled-components";
import {ReactComponent as ShuffleSvg} from '../svg/shuffle/shuffle.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonIcon = styled(PlayerButton)``;

export const PlayerButtonShuffle = styled(PlayerButtonIcon)`
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonShuffleSvg = styled(ShuffleSvg)`
width: 19px;
height: 12px;
fill: none;
stroke: ${props => props.theme === "light" ? "#B1B1B1" : "#696969"};
${PlayerButtonShuffle}:hover & {
    fill: none;
    stroke: ${props => props.theme === "light" ? "#707070" : "#ACACAC"};
    cursor: pointer;
  }
${PlayerButtonShuffle}:active & {
    fill: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    stroke: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    cursor: pointer;
  }
`;