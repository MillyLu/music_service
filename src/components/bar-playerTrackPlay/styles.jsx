import styled from "styled-components";


export const TrackPlay = styled.div`
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-orient: horizontal;
-webkit-box-direction: normal;
    -ms-flex-direction: row;
        flex-direction: row;
}
`;