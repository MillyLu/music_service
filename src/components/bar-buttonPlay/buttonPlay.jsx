
import { ReactComponent as StopSvg } from '../bar/stop.svg';
import {ReactComponent as PlaySvg} from '../svg/play/play.svg';
import * as Styled from './styles';

 export function ButtonPlay({ data, isPlaying, toggleAudio, aud, onLoadedMetadata, setPlaying, setEnd, reloop}) {

    return(
        <Styled.PlayerButtonPlay data-testid="play" isSelected={isPlaying} onClick={toggleAudio}>
                 <Styled.PlayerButtonPlaySvg >
                    {
                        isPlaying ? 
                        <StopSvg />
                            :
                        <PlaySvg />
                        
                    }
                     
                 </Styled.PlayerButtonPlaySvg>

                 <Styled.Audio loop={reloop}  onLoadedMetadata={onLoadedMetadata}  ref={aud} src={(data ? data?.track_file : '')} onPlay={()=>setPlaying(true)} onPause={()=>{setPlaying(false)}} onEnded={()=>{setEnd(true)}} autoPlay></Styled.Audio>
                         
                     
            
                </Styled.PlayerButtonPlay> 
    )
}