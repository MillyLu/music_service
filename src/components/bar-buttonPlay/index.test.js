/* eslint-disable no-undef */
import { render, screen, fireEvent } from "@testing-library/react";
import { ButtonPlay } from "./buttonPlay";
import { PlayerButtonPlay } from "./styles";



describe('<ButtonPlay />', () => {
    it("should render successfully and call onClick", async () => {

        const {container} = render(<ButtonPlay />);
        expect(container).toMatchSnapshot();


})

it("should render successfully and call onClick", async () => {
    const toggleAudio = jest.fn();
    render(<PlayerButtonPlay data-testid="play" onClick={toggleAudio} /> );
    fireEvent.click(screen.getByTestId('play'));
expect(toggleAudio).toBeCalledTimes(1)

    
})
})