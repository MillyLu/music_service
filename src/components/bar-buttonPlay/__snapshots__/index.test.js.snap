// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPlay /> should render successfully and call onClick 1`] = `
<div>
  <div
    class="sc-bcXHqe sc-gswNZR sc-dkrFOg cdqdtz dEeYDP TJkgy"
    data-testid="play"
  >
    <svg
      class="sc-hLBbgP fFHXSW"
    >
      <svg>
        play.svg
      </svg>
    </svg>
    <audio
      autoplay=""
      class="sc-eDvSVe gfFGNs"
      src=""
    />
  </div>
</div>
`;
